File: src/main/java/org/jcodec/containers/mp4/boxes/EncodedPixelBox.java
package org.jcodec.containers.mp4.boxes;

/**
 * This class is part of JCodec ( www.jcodec.org ) This software is distributed
 * under FreeBSD License
 * 
 * @author The JCodec project
 * 
 */
public class EncodedPixelBox extends ClearApertureBox {

    public static String fourcc() {
        return "enof";
    }

    public static EncodedPixelBox createEncodedPixelBox(int width, int height) {
        EncodedPixelBox enof = new EncodedPixelBox(new Header(fourcc()));
        enof.width = width;
        enof.height = height;
        return enof;
    }

    public EncodedPixelBox(Header atom) {
        super(atom);
    }
}

File: src/main/java/org/jcodec/containers/mp4/boxes/EncodedPixelBox.kt
package org.jcodec.containers.mp4.boxes

/**
 * This class is part of JCodec ( www.jcodec.org ) This software is distributed
 * under FreeBSD License
 *
 * @author The JCodec project
 */
class EncodedPixelBox(atom: Header) : ClearApertureBox(atom) {
    companion object {
        const val ENOF = "enof"
        @JvmStatic
        fun createEncodedPixelBox(width: Int, height: Int): EncodedPixelBox {
            val enof = EncodedPixelBox(Header(ENOF))
            enof.width = width.toFloat()
            enof.height = height.toFloat()
            return enof
        }
    }
}